cpp/include/cudf/lists/detail/gather.cuh

```diff
@@ -110,15 +119,20 @@ gather_data make_gather_data(cudf::lists_column_view const& source_column,
 
   // generate the base offsets
   rmm::device_uvector<int32_t> base_offsets = rmm::device_uvector<int32_t>(output_count, stream);
-  thrust::transform(rmm::exec_policy(stream),
-                    gather_map,
-                    gather_map + output_count,
-                    base_offsets.data(),
-                    [src_offsets, src_size, shift] __device__(int32_t index) {
-                      // if this is an invalid index, this will be a NULL list
-                      if (NullifyOutOfBounds && ((index < 0) || (index >= src_size))) { return 0; }
-                      return src_offsets[index] - shift;
-                    });
+  thrust::transform(
```

Comment:
I'm not a fan of using a column_device_view here.  Creating these isn't free (involves memory allocation on both cpu and gpu and some processing) which is going to get repeated as we march down the hierarchy.  If all we want to do is check the validity bits, just getting the bitmask pointer and calling `cudf::detail::bit_is_set()` is much lighter weight.

Reply:
Agreed. I've switched to use `bit_is_set()`. Please note that I've only changed this for `LIST`. In the `STRING` case, the `column_device_view` is constructed regardless.